### DISCUSSION TITLE
Nearby announcement enters into a loop

### DISCUSSION THREAD

**Discussion by airald** [devemux86/cruiser#300] [Q&A] [answered]
I was navigating to home. I have enabled announcements of gas stations and speed cameras. It works well on all of them except one speed camera. Each time Cruiser announces on specific speed camera, the message "speed camera in 700 meters" repeats continuously. First time I thought was something related with my phone this time. I stop navigation and message repeats and repeats until I close Cruiser.

The next day, the problem repeats again with same speed camera. This time I didn't touch anything, just wait. As I suspect, the message stops when Cruiser announced next turn (after 2 km approx).

I forced my phone to say something using Waze, the message from waze was played and Cruiser continue playing the announcement.

This morning I made the same route but in the other way to go to work. The message just was played ONE time.

I can not reproduce the problem in simulation, speed camera message is not played.

This afternoon I can do another test with some other settings...

I will sent to @devemux86 the used profile and the coordinates of the speed camera.

**Comment by devemux86** [comment]
@airald many thanks for the detailed report!

I can reproduce it and will check what happens in the algorithms.

> I can not reproduce the problem in simulation, speed camera message is not played.

They are announced, it is just that you have the simulation speed at maximum "5x"
and in simulation messages are interrupted when another is about to be announced.

---

Some quick hints to fix the issue yourself (until a proper fix comes in next Cruiser):
- In "Settings | Search | Advanced | Side of route" select "Right" (instead of default "Right + Left")
There are 2 speed cameras on each side of the road, Cruiser recognizes and announces both.

- Disable the "Settings | Voice guidance | Advanced | Current speed"

If you make one or both of these changes, the issue will be resolved.
Of course I will check what is happening to solve it internally as well!

**Reply by airald** [reply]
I suspect that could be related with point 1, but when Cruiser announces the speed camera when I go North->South, all is ok. 

I will check selecting right this afternoon.

Thank's!

**Reply by devemux86** [reply]
> when Cruiser announces the speed camera when I go North->South, all is ok.

You are right, this can be also reproduced. I will see what happens!

**Comment by devemux86** [comment]
@airald thanks very much for the report!

I found the problem, it will work properly in the next Cruiser.

**Reply by airald** [reply]
Tested with Settings | Search | Advanced | Side of route --> Right
Woks properly.

**Reply by devemux86** [reply]
@airald thanks for the feedback.

I will make a new Beta later with the fix, so it can work with all options.

**Comment by devemux86** [comment]
It should be fixed in:
- https://github.com/devemux86/cruiser/discussions/299

**Reply by airald** [reply]
The option to announce both right and left disappeared.

**Reply by devemux86** [reply]
@airald this option is for the search.
It is in "Settings | Search | Advanced | Side of route".

**Reply by airald** [reply]
you're true... I searched "side" but I forgot I use Cruiser in Catalan

**Comment by Ciutnik** [comment]
Unfortunately, beta v9, I experienced the same today. The voice announcement about two favorites, located very close to each other, looped endlessly. I thought my head would burst 😉 

The announcement was still "In 700 meters point A, in 700 meters point B" even though the distance was closing. I'm not sure what broke this loop - either a turn command or a reroute and manual recalculation.

**Reply by devemux86** [reply]
@Ciutnik thanks for the report!

Note: Favorites are different from POI and are processed with different algorithms.
The above fix was for the POI announcement.

If there is an issue with the favorites, can you send me the route and the favorites,
so I can check what happens?

Also did you have enabled the "Settings | Voice guidance | Advanced | Current speed"?

**Reply by Ciutnik** [reply]
> Also did you have enabled the "Settings | Voice guidance | Advanced | Current speed"?

Yes, I have it enabled. 
I am attaching the data set (including profile), maybe you can reproduce this strange behavior. Unfortunately, I don't remember which specific two points fell into the loop because they have strange names. These are amazing wagons, exhibits from the railway museum 🙂

[Loop.zip](https://github.com/devemux86/cruiser/files/12504047/Loop.zip)

**Reply by Ciutnik** [reply]
At home, in simulation mode, I can't reproduce this behavior.

If it can be useful, it's also track. 
[Loop_trk.zip](https://github.com/devemux86/cruiser/files/12504087/Loop_trk.zip)

**Reply by devemux86** [reply]
This seems to be caused by the "Current speed" setting.
Simulation uses a virtual (not real) speed, so it works a bit differently in that area.

**Reply by Ciutnik** [reply]
Is it my specific profile setting causing this? Maybe I made a mistake somewhere and I should avoid something?

I have attached the track above. The loop started 700 meters before the points and ended after turning off the route, near these points (U-turn).

**Reply by devemux86** [reply]
@Ciutnik for now just disable the "Settings | Voice guidance | Advanced | Current speed".

I will check what happens in the algorithms.

**Reply by devemux86** [reply]
@Ciutnik thanks for the files.

I still can't reproduce it. :slightly_smiling_face: 

Did you hear a loop of 2 different favorites or the same favorite?

**Reply by Ciutnik** [reply]
> Did you hear a loop of 2 different favorites or the same favorite?

Yes, it was two different points, spoken in a loop, one after the other. At least a dozen times, without a break, and each time "in 700 meters...".

**Reply by devemux86** [reply]
Do you remember which ones?
The first nearest seem to be the "414S" and "Wagon salonowy Ashx"

**Reply by Ciutnik** [reply]
I don't remember which ones they were. But it may be just these two, because one had a short symbol and the other a longer name. I was going north to south, so if these are the closest, they might be...

**Comment by devemux86** [comment]
We better remove the voice "Current speed" setting to avoid such complicated cases.

_Cruiser is the only navigator where we can adjust the voice distances._

This should be enough to make it work according to how each user wants.

**Reply by Ciutnik** [reply]
I guess that's a good decision. 

In the Garmin 276c, it was possible to set the TIME (instead of distance) to maneuver, in which a panel with turn instructions appeared. Clever. Probably a real-time average speed was used (similar to the always accurate calculation of the time of arrival). It worked great. In Cruiser, having profiles of various vehicles/activities with declared distances - roughly comes out the same 🙂

Because every situation is different, it really doesn't matter if we hear the navigation announcement 300 or 320 meters before the maneuver, much less 3000 or 3200 meters 🙂

**Reply by 0709wiwiwi** [reply]
> In the Garmin 276c, it was possible to set the TIME (instead of distance) to maneuver, in which a panel with turn instructions appeared. Clever. Probably a real-time average speed was used (similar to the always accurate calculation of the time of arrival). It worked great.

Locus is using the same  "by avg speed" principle.  I do use this for touristic cycling ( 18kmh avg) and usually set a single announcement with display on/off mode at 15 sec before the turns 
As long you are operating with higher than walking speed it works fine. Announcement tolerances 10 m or even 20 m are unproblematic.
However at slow speeds like a walk in woods or in cities, because of the gps position vs open street map tolerances the announcements often come too late.  A minimum distance announcement should so be combined with such timed announcements. This was requested but it not implemented :-((

The announcement by distance trigger advantage is to be simple and everyone does understand.  
The announcement setting by time before the maneuver often must be explained to the newbies.

**Reply by devemux86** [reply]
Note that before we introduce voice distance settings in Cruiser,
I had implemented voice guidance based on timing using speed.

It still exists in my code and should be similar to what you describe.

But it wasn't easy for users to understand,
so we changed it to the current freely customizable voice distances. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
@0709wiwiwi 
> The announcement by distance trigger advantage is to be simple and everyone does understand

@devemux86 
> But it wasn't easy for users to understand

We all agree on this topic. Clever solutions are very cool, but unusual situations can be less predictable. Keep it simple, reliable and unambiguous.

**Comment by devemux86** [comment, marked as answer]
Should be fixed in [Cruiser 3.0.32](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
